src/test/java/pl/piekoszek/gorskimatches/controllers/ChallengeControllerTest.java

```diff
@@ -0,0 +1,191 @@
+
+@SpringBootTest
+@AutoConfigureMockMvc
+public class ChallengeControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private TokenService tokenService;
+
+    @Test
+    void shouldReturnUuid() throws Exception {
+        mockMvc.perform(get("/api/challenge/generate"))
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    void shouldSaveScoreOfUserWithEmail() throws Exception {
+        var uuid = getUuid();
+
+        mockMvc.perform(post("/api/challenge/score")
+                        .header("Authorization", "Bearer " + getToken())
+                        .content(getJsonString(uuid))
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    void shouldSaveScoreOfUserWithoutEmail() throws Exception {
+        var uuid = getUuid();
+
+        mockMvc.perform(post("/api/challenge/score")
+                        .header("Authorization", "Bearer " + getToken())
+                        .content(getJsonString(uuid))
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+
+        mockMvc.perform(post("/api/challenge/score")
+                        .content("""
+                                {
+                                 "uuid": "%s",
+                                 "score" : "3",
+                                 "time" : "14.21"
+                                }
+                                """.formatted(uuid))
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    void shouldReturnResultForNonRegisteredUser() throws Exception {
+        var uuid = getUuid();
+
+        mockMvc.perform(post("/api/challenge/score")
+                        .header("Authorization", "Bearer " + getToken())
+                        .content(getJsonString(uuid))
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+
+        mockMvc.perform(post("/api/challenge/score")
+                        .content("""
+                                {
+                                 "uuid": "%s",
+                                 "score" : "3",
+                                 "time" : "14.21"
+                                }
+                                """.formatted(uuid))
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+
+        mockMvc.perform(get("/api/challenge/resultForNonRegistered/{uuid}", uuid))
+                .andExpect(status().isOk());
```

Comment:
You should assert response body contents, not only status. This applies also to other requests where response is useful to user.